### PR TITLE
pd, openapi(ticdc): retry one reset by peer when kill the pd instance, fix healthy panic

### LINKED ISSUE #7067: kill -9 PD non-leader instance, cause all 2 cdc instances restarted.
### What did you do?

2 pd, 2 ticdc.

`kill -9` pd non leader instance.

### What did you expect to see?

ticdc should works normally

### What did you see instead?

all 2 cdc nodes restart

```
[2022/09/14 18:47:45.182 +08:00] [WARN] [client.go:97] ["etcd RPC failed"] [RPC=Txn] [error="rpc error: code = Unavailable desc = error reading from server: read tcp 10.244.3.5:49780->10.244.9.155:2379: read: connection reset by peer"]
[2022/09/14 18:47:45.182 +08:00] [INFO] [client.go:235] ["WatchWithChan exited"] [role=owner]
[2022/09/14 18:47:47.354 +08:00] [WARN] [server.go:302] ["etcd health check error"] [endpoint=http://lingjin-1.jinling.tispace:2379] [error="Get \"http://lingjin-1.jinling.tispace:2379/pd/api/v1/health/\": dial tcp 10.244.9.155:2379: connect: connection refused"] [errorVerbose="Get \"http://lingjin-1.jinling.tispace:2379/pd/api/v1/health/\": dial tcp 10.244.9.155:2379: connect: connection refused\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.Trace\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/juju_adaptor.go:15\ngithub.com/pingcap/tiflow/pkg/pdutil.(*pdAPIClient).Healthy\n\tgithub.com/pingcap/tiflow/pkg/pdutil/api_client.go:224\ngithub.com/pingcap/tiflow/cdc/server.(*server).etcdHealthChecker\n\tgithub.com/pingcap/tiflow/cdc/server/server.go:301\ngithub.com/pingcap/tiflow/cdc/server.(*server).run.func2\n\tgithub.com/pingcap/tiflow/cdc/server/server.go:324\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:75\nruntime.goexit\n\truntime/asm_amd64.s:1571"]
```

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
master
```

### LINKED ISSUE #7094: cdc open api healthy panic since there is not owner
### What did you do?

call `healthy`

### What did you expect to see?

works normal

### What did you see instead?

```
[31m2022/09/16 12:00:57 [Recovery] 2022/09/16 - 12:00:57 panic recovered:
GET /api/v1/health HTTP/1.1
Host: lingjin-3.jinling.tispace:8300
Accept-Encoding: gzip
User-Agent: Go-http-client/1.1


runtime error: invalid memory address or nil pointer dereference
runtime/panic.go:220 (0x44ecf5)
runtime/signal_unix.go:818 (0x44ecc5)
[github.com/pingcap/tiflow/cdc/api/v1/api.go:877](http://github.com/pingcap/tiflow/cdc/api/v1/api.go:877) (0x2c354d7)
[github.com/gin-gonic/gin@v1.7.4/context.go:165](http://github.com/gin-gonic/gin@v1.7.4/context.go:165) (0x2bfabb1)
[github.com/pingcap/tiflow/cdc/api/middleware/middleware.go:64](http://github.com/pingcap/tiflow/cdc/api/middleware/middleware.go:64) (0x2bfab9d)
[github.com/gin-gonic/gin@v1.7.4/context.go:165](http://github.com/gin-gonic/gin@v1.7.4/context.go:165) (0x2bfa279)
[github.com/pingcap/tiflow/cdc/api/middleware/middleware.go:38](http://github.com/pingcap/tiflow/cdc/api/middleware/middleware.go:38) (0x2bfa25c)
[github.com/gin-gonic/gin@v1.7.4/context.go:165](http://github.com/gin-gonic/gin@v1.7.4/context.go:165) (0x2bfb081)
[github.com/pingcap/tiflow/cdc/api/middleware/middleware.go:103](http://github.com/pingcap/tiflow/cdc/api/middleware/middleware.go:103) (0x2bfafd2)
[github.com/gin-gonic/gin@v1.7.4/context.go:165](http://github.com/gin-gonic/gin@v1.7.4/context.go:165) (0x126e4e1)
[github.com/gin-gonic/gin@v1.7.4/recovery.go:99](http://github.com/gin-gonic/gin@v1.7.4/recovery.go:99) (0x126e4cc)
[github.com/gin-gonic/gin@v1.7.4/context.go:165](http://github.com/gin-gonic/gin@v1.7.4/context.go:165) (0x126d35d)
[github.com/gin-gonic/gin@v1.7.4/gin.go:489](http://github.com/gin-gonic/gin@v1.7.4/gin.go:489) (0x126cfe5)
[github.com/gin-gonic/gin@v1.7.4/gin.go:445](http://github.com/gin-gonic/gin@v1.7.4/gin.go:445) (0x126cb44)
net/http/server.go:2916 (0x726fba)
net/http/server.go:1966 (0x721ef6)
runtime/asm_amd64.s:1571 (0x46c3c0)
[0m
```

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
master
```

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #7067, close #7094, ref #4757

### What is changed and how it works?

* retry on receive `RST`, which was caused by the PD instance shut down.

image

#### Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: pkg/errorutil/ignore.go [hunk 1]
```diff
@@ -84,5 +84,10 @@ func IsRetryableEtcdError(err error) bool {
 	if strings.Contains(etcdErr.Error(), "received prior goaway: code: NO_ERROR") {
 		return true
 	}
+
+	// this may happen if the PD instance shutdown by `kill -9`, no matter the instance is the leader or not.
+	if strings.Contains(etcdErr.Error(), "connection reset by peer") {
```
Comment (overvenus) [inline pkg/errorutil/ignore.go:89]: It maybe not retryable in cases do operations that are not idempotent.

Comment (overvenus) [reply pkg/errorutil/ignore.go:89]: Could you add the check to pkg/etcd?

**Review by overvenus — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**